src/record/reader.rs

```diff
@@ -297,7 +298,24 @@ impl Reader {
 
   /// Reads current record as `Row` from the reader tree.
   /// Automatically advances all necessary readers.
+  /// This must be called on the root level reader (i.e., for Message type).
+  /// Otherwise, it will panic.
   fn read(&mut self) -> Row {
```

Comment:
When do we call `read`? I am just a bit confused:)

Reply:
The `read` is called on `next()` for both `RowIter` and `ReaderIter`.